### PR TITLE
make gci linter fixable

### PR DESCRIPTION
**Pull request by vasyharan**
Fixes https://github.com/golangci/golangci-lint/issues/2604

### CODE REVIEW

**Comment by vasyharan** [comment]
sorry about the noise. i was figuring out how to get the workflow to run on my fork to address test failures. this should be addressed now (https://github.com/vasyharan/golangci-lint/pull/1 had a successful run)

**Comment by ldez** [comment]
We have to approve the run of the workflow because it's your first contribution.
So the CI will not start, it will start when we will allow it.

https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

**Comment by vasyharan** [comment]
anything i can help with / do to unblock this?

**Comment by barweiss** [comment]
Hi @vasyharan, @ldez!
Is there any way to help with this issue?

**Comment by SpaWn2KiLl** [comment]
@ldez can you have a look on this please? :)

**Comment by daixiang0** [comment]
@vasyharan please fix conflicts.

File: pkg/golinters/gci.go [hunk 1]
```diff
@@ -15,34 +20,94 @@ import (
...
+						issues = append(issues, goanalysis.NewIssue(&is[i], pass))
+					}
+				}
+			}
+
+			if len(issues) == 0 {
+				return nil, nil
+			}
+
+			mu.Lock()
```
Comment (daixiang0) [inline pkg/golinters/gci.go:104]: Why need lock here?

**Comment by ldez** [comment]
closed in favor of #2892

**Pull request closed without merging**